libminifi/src/core/logging/LoggerConfiguration.cpp

```diff
@@ -294,26 +295,27 @@ std::shared_ptr<spdlog::logger> LoggerConfiguration::get_logger(std::shared_ptr<
   return spdlog::get(name);
 }
 
-std::shared_ptr<spdlog::sinks::sink> LoggerConfiguration::create_syslog_sink() {
+spdlog::sink_ptr LoggerConfiguration::create_syslog_sink() {
 #ifdef WIN32
   return std::make_shared<internal::windowseventlog_sink>("ApacheNiFiMiNiFi");
 #else
-  return std::make_shared<spdlog::sinks::syslog_sink>("ApacheNiFiMiNiFi");
+  return std::dynamic_pointer_cast<spdlog::sinks::sink>(spdlog::syslog_logger_mt("ApacheNiFiMiNiFi", 0, LOG_USER, false));
 #endif
 }
 
-std::shared_ptr<spdlog::sinks::sink> LoggerConfiguration::create_fallback_sink() {
+spdlog::sink_ptr LoggerConfiguration::create_fallback_sink() {
   if (utils::Environment::isRunningAsService()) {
     return LoggerConfiguration::create_syslog_sink();
   } else {
-    return spdlog::sinks::stderr_sink_mt::instance();
+    return std::dynamic_pointer_cast<spdlog::sinks::sink>(std::make_shared<spdlog::sinks::stderr_sink_mt>());
   }
 }
 
 std::shared_ptr<internal::LoggerNamespace> LoggerConfiguration::create_default_root() {
   std::shared_ptr<internal::LoggerNamespace> result = std::make_shared<internal::LoggerNamespace>();
   result->sinks = std::vector<std::shared_ptr<spdlog::sinks::sink>>();
-  result->sinks.push_back(spdlog::sinks::stderr_sink_mt::instance());
+  auto default_sink = std::make_shared<spdlog::sinks::stderr_sink_mt>();
+  result->sinks.emplace_back(std::move(default_sink));
```

Comment:
Instead of the 3 lines above:
```
result->sinks = { std::make_shared<spdlog::sinks::stderr_sink_mt>(); };
```

Reply:
That is a valid and useful refactor. I did not check the context on doing the update.